torchvision/prototype/transforms/_augment.py

```diff
@@ -0,0 +1,149 @@
+import math
+import numbers
+import warnings
+from typing import Any, Dict, Tuple
+
+import torch
+from torchvision.prototype.transforms import Transform
+from torchvision.prototype.utils._internal import query_recursively
+
+from . import functional as F
+
+
+class RandomErasing(Transform):
+    _DISPATCHER = F.erase
+
+    def __init__(
+        self,
+        p: float = 0.5,
+        scale: Tuple[float, float] = (0.02, 0.33),
+        ratio: Tuple[float, float] = (0.3, 3.3),
+        value: float = 0,
+    ):
+        super().__init__()
+        if not isinstance(value, (numbers.Number, str, tuple, list)):
+            raise TypeError("Argument value should be either a number or str or a sequence")
+        if isinstance(value, str) and value != "random":
+            raise ValueError("If value is str, it should be 'random'")
+        if not isinstance(scale, (tuple, list)):
+            raise TypeError("Scale should be a sequence")
+        if not isinstance(ratio, (tuple, list)):
+            raise TypeError("Ratio should be a sequence")
+        if (scale[0] > scale[1]) or (ratio[0] > ratio[1]):
+            warnings.warn("Scale and ratio should be of kind (min, max)")
+        if scale[0] < 0 or scale[1] > 1:
+            raise ValueError("Scale should be between 0 and 1")
+        if p < 0 or p > 1:
+            raise ValueError("Random erasing probability should be between 0 and 1")
+        # TODO: deprecate p in favor of wrapping the transform in a RandomApply
```

Comment:
This TODO is probably left from earlier when you used more abstraction. Is this still relevant given the changes?

Reply:
Yes, this is still relevant. The idea is to remove all probability arguments from the transforms in favor of wrapping them in a `RandomApply` transform. In this example this means

```py
transforms.RandomErasing(p=..., ...)
```

would change to

```py
transforms.RandomApply(
    transforms.RandomErasing(...),
    p=...,
)
```

Another example is to no longer have a `RandomHorizontalFlip`. The term "Random" in the name of a transform should mean "random parameters sampled at runtime".